src/rxreact.tsx

```diff
@@ -22,7 +22,7 @@ function withViewModelFactory<S, A, P>(
     WrappedComponent: React.ComponentType<T>
   ): React.ComponentClass<Difference<T, S & ActionMap<A>> & P> {
     return class ConnectState extends React.Component<Difference<T, S & ActionMap<A>> & P, S> {
-      subscription: Subscription | undefined;
+      subscriptions: Subscription[] = [];
```

Comment:
treat the users's subscriptions and the overall subscription as an array of stuff to unsub from on unmount